messaging/src/test/java/org/axonframework/queryhandling/DefaultReactorQueryGatewayTest.java

```diff
@@ -0,0 +1,780 @@
+                                                                                                            Integer.class);
+        SubscriptionQueryResult<Integer, Integer> result = monoResult.block();
+        assertNotNull(result);
+        StepVerifier.create(result.initialResult())
+                    .verifyError(RuntimeException.class);
+    }
+
+    @Test
+    void testSubscriptionQueryFailsRetryInitialDispatchQuery() throws Exception {
+
+        doThrow(new RuntimeException(":(")).when(queryBus).subscriptionQuery(any(), any(), anyInt());
+
+        Mono<SubscriptionQueryResult<Integer, Integer>> monoResult = reactiveQueryGateway.subscriptionQuery(6,
+                                                                                                            Integer.class,
+                                                                                                            Integer.class)
+                                                                                         .retry(5);
+
+        StepVerifier.create(monoResult)
+                    .verifyError(RuntimeException.class);
+
+        verify(queryBus, times(6)).subscriptionQuery(any(), any(), anyInt());
+    }
+
+    @Test
+    void testSubscriptionQueryFailsRetryInitialResult() throws Exception {
+        Mono<SubscriptionQueryResult<Integer, Integer>> monoResult = reactiveQueryGateway.subscriptionQuery(6,
+                                                                                                            Integer.class,
+                                                                                                            Integer.class);
+
+        SubscriptionQueryResult<Integer, Integer> result = monoResult.block();
+        assertNotNull(result);
+        StepVerifier.create(result.initialResult().retry(5))
+                    .verifyError(RuntimeException.class);
+
+        verify(queryMessageHandler3, times(6)).handle(any());
+    }
+
+    @Test
+    void testSubscriptionQuerySingleInitialResultAndUpdates() {
+        Flux<String> result = reactiveQueryGateway.subscriptionQuery("6", String.class);
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.execute(() -> {
+            try {
+                countDownLatch.await();
+            } catch (InterruptedException e) {
+                fail(e.getMessage());
+            }
+            queryUpdateEmitter.emit(String.class, q -> true, "1");
+            queryUpdateEmitter.emit(String.class, q -> true, "2");
+            queryUpdateEmitter.emit(String.class, q -> true, "3");
+            queryUpdateEmitter.emit(String.class, q -> true, "4");
+            queryUpdateEmitter.emit(String.class, q -> true, "5");
+            queryUpdateEmitter.complete(String.class, p -> true);
```

Comment:
Curious/nit: Why is the lambda variable a `p` for the `QueryUpdateEmitter#complete` call? Shouldn't that be a `q` too?

Reply:
changed to `p`